### PR TITLE
feat(tezos): upgrade to taquito v8

### PR DESCRIPTION
**Pull request by shekhar-shubhendu**
#3500

### CODE REVIEW

**Comment by shekhar-shubhendu** [comment]
Changes for delphinet ganache-core https://github.com/trufflesuite/ganache-core/pull/660 that will fix the failing builds

**Comment by eggplantzzz** [comment]
@shekhar-shubhendu Thanks for the PR! I just wanted to let you know that you can remove the configuration for the Node 8 builds in CI. You don't have to worry about getting them to pass.

**Comment by shekhar-shubhendu** [comment]
@eggplantzzz Thanks for the tip. I've updated the nodejs version to 10 (as per master).
But the builds will fail until we release a new version of ganache-core@tezos and update it here as the currently installed version of ganache-core is trying to patch the older version of `@taquito/taquito` and it fails to find it as this PR has updated that.

**Comment by shekhar-shubhendu** [comment]
@eggplantzzz I've updated the PR with latest ganache-core@tezos release and also bumped taquito slightly.
But I'm unable to fix the CI issues here and will need some help.

**Comment by eggplantzzz** [comment]
@shekhar-shubhendu Hmm, I'm not sure I can provide much help on this as I don't have too much familiarity with the Tezos stuff. I'll see if I can think of some way to move this forward and get back to you if I get any ideas. Thanks!

**Comment by shekhar-shubhendu** [comment]
> @shekhar-shubhendu Hmm, I'm not sure I can provide much help on this as I don't have too much familiarity with the Tezos stuff. I'll see if I can think of some way to move this forward and get back to you if I get any ideas. Thanks!

I don't think it is the tezos part. But the CI itself seems to be failing. The tests run fine if I run them locally. I'll spend sometime on it tomorrow morning and see if I figure it out. If you find something too then let me know and I'll try to fix it here.

**Comment by eggplantzzz** [comment]
I'll also restart the CI just in case there was some hiccup that caused it to fail.

**Comment by haltman-at** [comment]
Hey, I took a look at why the tests are failing, and it does seem to be unrelated to Tezos.  It seems to be something we fixed a while back; I was going to ask you to merge in `develop`, but that obviously could be troublesome.  So (assuming you don't want to attempt that) could you just merge in #3210?  I think that should fix the CI problem.  Thank you!

**Comment by shekhar-shubhendu** [comment]
@haltman-at thanks for looking into this. I can try to merge develop into this first (will skip if I think it is too much) if you prefer that?

**Comment by haltman-at** [comment]
Up to you!  @gnidan was of the opinion that it would probably be too much trouble to be worth it.

**Comment by shekhar-shubhendu** [comment]
to enable tezos specific tests we need now these 2 PRs
I'm not sure how useful the current tests are though :)

https://github.com/truffle-box/tezos-smartpy-example-box/pull/1
https://github.com/truffle-box/tezos-example-box/pull/9

**Comment by haltman-at** [comment]
@shekhar-shubhendu Could you run `yarn` on your branch and commit any changes to `yarn.lock` that result?  This PR is currently failing our yarncheck job which tries to make sure all necessary changes to `yarn.lock` get committed.  Thank you!  (I haven't looked at why the other test might be failing.)

**Comment by shekhar-shubhendu** [comment]
> @shekhar-shubhendu Could you run `yarn` on your branch and commit any changes to `yarn.lock` that result? This PR is currently failing our yarncheck job which tries to make sure all necessary changes to `yarn.lock` get committed. Thank you! (I haven't looked at why the other test might be failing.)

aha. I did not knew what that was and why it was failing. I will do that. Thanks for tip 🙂

**Comment by haltman-at** [comment]
Blech, I see the `compile-solidity` tests are still failing.  I'm going to see if I can find any other relevant PRs there to merge in? :-/  Otherwise you can always try to merge in `develop` of course, but...

**Comment by haltman-at** [comment]
OK, I think I found it -- try merging in #3252 if possible.  That should be easier than merging in all of `develop`...

**Comment by shekhar-shubhendu** [comment]
@haltman-at the build for `db` package seems to get stuck at `ttsc --project tsconfig.codegen.json`
is there something that I need to do to fix this?

**Comment by haltman-at** [comment]
No idea; I don't work on `@truffle/db`.  @fainashalts, @gnidan, do you have any ideas here?

**Comment by shekhar-shubhendu** [comment]
> No idea; I don't work on `@truffle/db`. @fainashalts, @gnidan, do you have any ideas here?

Hmmm. It seems to be happening only in this branch. And I have latest from develop.

**Comment by eggplantzzz** [comment]
Hmm, it is possible that taquito has gotten too big for our webpack config. You could maybe try allocating more memory for that process on building? If you try it, let us know what kind of results you get and we'll see if there is any other advice we can provide. Thanks!

**Comment by shekhar-shubhendu** [comment]
@eggplantzzz 
It builds cleanly for me on node 10(and 12, 14 too). 
This is what I have:

- node v10.23.0
- npm v6.14.8
- yarn v1.22.10
- lerna v3.22.1

**Comment by haltman-at** [comment]
Btw, I should note that although this isn't the main problem you're having right now, but what with the recent release of Solidity 0.8.0 you're going to need to pull in #3612 (or `develop` as a whole) to fix problems that will cause.

**Comment by shekhar-shubhendu** [comment]
@gnidan I've removed the develop merge (and all that came after that) from the branch

Also, please take a look at below PRs too
[truffle-box/tezos-smartpy-example-box#1](https://github.com/truffle-box/tezos-smartpy-example-box/pull/1)
[truffle-box/tezos-example-box#9](https://github.com/truffle-box/tezos-example-box/pull/9)

And these are the ones from ganache
https://github.com/trufflesuite/ganache-cli/pull/808
https://github.com/trufflesuite/ganache-core/pull/677

**Comment by haltman-at** [comment]
Hm, well, you'll need to do something to get those latest-Solidity-based tests to the right version...

**Comment by shekhar-shubhendu** [comment]
@haltman-at In a meeting with @gnidan (about a week ago) it was discussed that I'll remove the merge from develop and other stuff and only keep the tezos related things in this PR.

**Comment by haltman-at** [comment]
Oh, hm. How is this going to get passing, then...? (@gnidan?)

**Comment by gnidan** [comment]
I vote: just delete the failing Solidity tests. We should get this green so we make sure Tezos stuff isn't breaking, but this branch doesn't need to ensure non-Tezos stuff at all.

**Comment by andrewpaulicek** [comment]
guys Edo protocol will be the mainnet protocol in Tezos in 10 days. people are getting constraint by having an external dependency on Truffle. can we please move this forward and get it running?
@gnidan

**Comment by shekhar-shubhendu** [comment]
for `TEZOS=true yarn ci` to work we need to merge first

[truffle-box/tezos-smartpy-example-box#1](https://github.com/truffle-box/tezos-smartpy-example-box/pull/1)
[truffle-box/tezos-example-box#9](https://github.com/truffle-box/tezos-example-box/pull/9)

**Comment by shekhar-shubhendu** [comment]
@gnidan it would be nice if someone can review the things this PR does and provide some feedback.
From what I have tested the contract deployment and testing work fine but there seems to be a race condition in migrations.
I think this is 'caused by `setWallet` being called in the `provisioner` but I may be wrong.

File: packages/deployer/src/deployment.js [hunk 1]
```diff
@@ -330,6 +331,11 @@ class Deployment {
           eventArgs.estimateError = err;
         }
 
+        /**
+         * Required as this sets the wallet for each migration
+         */
+        tezos && contract.interfaceAdapter.setWallet(self);
```
Comment (davidmurdoch) [inline packages/deployer/src/deployment.js:337]: I think this needs to be

```suggestion
        tezos && (await contract.interfaceAdapter.setWallet(self));
```

@gnidan your thoughts?

File: packages/provisioner/index.js [hunk 2]
```diff
@@ -17,8 +16,12 @@ var provision = function(abstraction, options) {
     abstraction.ens = options.ens;
   }
 
+  if (abstraction.interfaceAdapter.tezos) {
+    abstraction.interfaceAdapter.setWallet(options);
```
Comment (davidmurdoch) [inline packages/provisioner/index.js:20]: I don't think this call can happen here, at least not how this `provision` function is set up now (not async). Not sure where the best place for this is.

File: packages/deployer/src/deployment.js (hunk 1 above)
Comment (gnidan) [reply packages/deployer/src/deployment.js:337]: Looks right to me

File: packages/provisioner/index.js (hunk 2 above)
Comment (shekhar-shubhendu) [reply packages/provisioner/index.js:20]: So if we do not do this here, the contract is unable to find a wallet.
Is there a place where we can set up the wallet when contract is instantiated?

File: packages/deployer/src/deployment.js (hunk 1 above)
Comment (shekhar-shubhendu) [reply packages/deployer/src/deployment.js]: Applied.

**Comment by shekhar-shubhendu** [comment]
@gnidan @davidmurdoch do you know how to resolve this issue 🙏🏽 
https://github.com/trufflesuite/truffle/pull/3510/checks?check_run_id=1989780309#step:6:98

**Comment by davidmurdoch** [comment]
> @gnidan @davidmurdoch do you know how to resolve this issue 🙏🏽
> Happens only in nodejs 14
> [#3510 (checks)](https://github.com/trufflesuite/truffle/pull/3510/checks?check_run_id=1989780309#step:6:98)

The tezos branch split from develop before we support for Node 14, which would be why the tests fail for node 14. You can either revert the change adding Node 14 to GitHub Actions or just disable the Ethereum-related tests that are failing.

**Comment by eggplantzzz** [comment]
Closing for maintenance.

**Pull request closed without merging**